### ISSUE TITLE
Add penalty in routing for barrier

### ISSUE THREAD

**Issue by Jean-BaptisteC** [organicmaps/organicmaps#4825]
**Describe the issue**
OM use highway exits too late and increase numbers of kilometers

**Steps to reproduce**
1. Search this place https://www.openstreetmap.org/way/1091347200
2. Route from
3. Search this place https://www.openstreetmap.org/node/4453337126#map=16/49.0455/6.6038
4. OM calculate routing 36 min - 52km
5. OM used this road https://www.openstreetmap.org/way/227940816 to leave highway (used this exit highway is not perfect, because he have gate on road and we used have=service)

**Expected behavior**
https://www.openstreetmap.org/directions?engine=graphhopper_car&route=49.1202%2C6.1684%3B49.0455%2C6.6038#map=12/49.0930/6.3861

**System information:**
 - Operating system and its version: Android 13
 - Organic Maps version: 2023.03.05-5-Google
 - Device Model: Android 13

CC @vng

**Comment by vng** [comment]
Well, yes.
- Barrier (https://www.openstreetmap.org/node/5039990764). We treat permissive as Yes. Should add some penalty anyway.
- Pass-through service road is allowed in France profile. Is it ok?

**Comment by Jean-BaptisteC** [comment]
Yes, but i don't understand why OM doesn't used [this exit highway](https://www.openstreetmap.org/node/29765355) because with this route, we lost 10 km

**Comment by vng** [comment]
Calculated ETA when following A4 till the gas station is better with 30 seconds :)

**Comment by Jean-BaptisteC** [comment]
OK, i understand
But do you think it's good idea to allow access permissive in routing (for gate), because with permissive, access can be revoke by owner and we don't have any information to know is access is always permit
https://wiki.openstreetmap.org/wiki/Tag:access%3Dpermissive

**Comment by biodranik** [comment]
If permissive access is revoked, then OpenStreetMap.org is updated, and OM works as usual.

**Comment by Jean-BaptisteC** [comment]
Yes, you right
We need add penalty on barrier
@vng does we have a penalty on toll gate ?

[Link about penalty used on barrier on OSMAnd](https://github.com/osmandapp/OsmAnd-resources/blob/db3d0756624c6173c37912b036e391ac2bb3c472/routing/routing.xml#L881)

**Comment by Jean-BaptisteC** [comment]
Problem was fixed on the latest version
